app/routes/task_routes.py

```diff
@@ -1 +1,156 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, abort, make_response, request, Response
+from datetime import date
+from app.models.task import Task
+from sqlalchemy import func, desc
+from .route_utilities import validate_model, model_from_request
+from ..models.goal import Goal
+from ..db import db
+import os
+import requests
+
+bp = Blueprint("task_bp", __name__, url_prefix='/tasks')
+
+SLACK_URL = 'https://slack.com/api/chat.postMessage'
+
+@bp.post("")
+def create_task():
+
+    new_task = model_from_request(cls=Task, request=request)
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return {"task": new_task.to_dict()}, 201
+
+
+@bp.get("")
+def get_all_tasks():
+
+    title_param = request.args.get("title")
+    description_param = request.args.get("description")
+    sort = request.args.get("sort")
+
+    query = sort_tasks_by(sort)
+
+    if title_param:
+        query = query.where(Task.title.ilike(f"%{title_param}%"))
+
+    if description_param:
+        query = query.where(Task.description.ilike(f"%{description_param}%"))
+
+    tasks = db.session.scalars(query)
+
+    tasks_response = []
+
+    for task in tasks:
+        tasks_response.append(task.to_dict())
+
+    return tasks_response
+
+@bp.get("/<task_id>")
+def get_single_task(task_id):
+
+    task = validate_model(Task, task_id)
+
+    return {
+        "task": task.to_dict()
+    }
+
+@bp.patch("/<task_id>/mark_complete")
+def mark_task_complete(task_id):
+
+    task = validate_model(Task, task_id)
+
+    task = change_status(task, True)
+
+    db.session.commit()
+
+    api_key = os.environ.get('SLACK_API_KEY')
+
+    headers = {
+        "Authorization": f"Bearer {api_key}",
+        "Content-Type": "application/json"
+    }
+
+    slack_message = {
+        "channel": "test-slack-api",
+        "text": f"Someone just completed the task {task.title}",
+    }
+
+    result = requests.post(url=SLACK_URL, json=slack_message, headers=headers)
```

Comment:
Prefer all this logic to live in a helper function, maybe something like `call_slack_api` to keep this route more concise and single-responsibility.

Also notice that `result` is not accessed so we don't need a variable to reference the return value from `requests.post`